### ISSUE TITLE
Failed to create Navigation Schematic 

### ISSUE THREAD

**Issue by abhiindian** [angular/material2#11521]
#### Bug:

Failed to create Navigation Schematic using Angular CLI and causing error while running Angular application using ng serve.

#### What is the expected behavior?
The navigation schematic should be displayed while running or building angular application using ng serve or ng build

#### What is the current behavior?

Thowing exception on ng serve.

![image](https://user-images.githubusercontent.com/7735653/40571637-3f48ca80-60ba-11e8-92b9-9798b672b4a7.png)

> 
> ERROR in src/app/navigation/navigation.component.ts(13,3): error TS2322: Type 'Observable<boolean>' is not assignable to type 'Observable<BreakpointState>'.
>   Type 'boolean' is not assignable to type 'BreakpointState'

#### What are the steps to reproduce?

ng new Blog

cd Blog

ng add @angular/material

ng generate @angular/material:material-nav --name navigation

remove all html code from app.component.html and add  `<navigation></navigation>`.

#### What is the use-case or motivation for changing an existing behavior?

Create Angular Material Navigation using Angular CLI.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

```js
{
  "name": "blog",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^6.0.3",
    "@angular/common": "^6.0.3",
    "@angular/compiler": "^6.0.3",
    "@angular/core": "^6.0.3",
    "@angular/forms": "^6.0.3",
    "@angular/http": "^6.0.3",
    "@angular/material": "^6.1.0",
    "@angular/platform-browser": "^6.0.3",
    "@angular/platform-browser-dynamic": "^6.0.3",
    "@angular/router": "^6.0.3",
    "core-js": "^2.5.4",
    "rxjs": "^6.0.0",
    "zone.js": "^0.8.26",
    "@angular/cdk": "^6.1.0"
  },
  "devDependencies": {
    "@angular/compiler-cli": "^6.0.3",
    "@angular-devkit/build-angular": "~0.6.5",
    "typescript": "~2.7.2",
    "@angular/cli": "~6.0.5",
    "@angular/language-service": "^6.0.3",
    "@types/jasmine": "~2.8.6",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "~4.2.1",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~1.7.1",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~1.4.2",
    "karma-jasmine": "~1.1.1",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.3.0",
    "ts-node": "~5.0.1",
    "tslint": "~5.9.1"
  }
}
```
![image](https://user-images.githubusercontent.com/7735653/40571680-0a6de1a0-60bb-11e8-8adb-071e2144c662.png)

**Comment by juristr** [comment]
Yep, just found the same.

In the component, it should be like this:

```
export class ShellComponent {
  isHandset$: Observable<BreakpointState> = this.breakpointObserver.observe(
    Breakpoints.Handset
  );
  // .pipe(
  //   map(result => result.matches)
  // );

  constructor(private breakpointObserver: BreakpointObserver) {}
}
```

Moreover, in the according template, the `async` needs to be wrapped with `( ... )` like

```
[mode]="(isHandset$ | async) ? 'over' : 'side'" ...>
```

After that, the app should be running. However, there's another issue. The overlay doesn't expand over the entire page, but rather just keeps being over the toolbar and content you added:

![image](https://user-images.githubusercontent.com/542458/40576312-4064cf56-60f4-11e8-8c6a-a83c3d6656c4.png)

This leads to the strange behavior that the side-nav obviously just closes when you click exactly on that toolbar where the overlay is. I'd change this behaviour to stretch to the entire page

**Comment by juristr** [comment]
@rafaelss95 perfect. Did search for the issue but couldn't find it.

However the overlay issue still exists, don't you think? IMHO it should expand the entire page

**Comment by sudhaparayil** [comment]
I am faceing the same issue..and solve it

app.module.ts
-------------------------------
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NavigationComponent } from './components/dashboard/navigation.component';
import { LayoutModule } from '@angular/cdk/layout';
import { MatToolbarModule, MatButtonModule, MatSidenavModule, MatIconModule, MatListModule } from '@angular/material';
const appRoutes: Routes = [
  { path: '', component: NavigationComponent },

];
@NgModule({
  declarations: [
    AppComponent,
    NavigationComponent 
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    RouterModule.forRoot(appRoutes),
    LayoutModule,
    MatToolbarModule,
    MatButtonModule,
    MatSidenavModule,
    MatIconModule,
    MatListModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }




navigation.component.html
---------------------------------------
<mat-sidenav-container class="sidenav-container">
  <mat-sidenav
    #drawer
    class="sidenav"
    fixedInViewport="true"
    [attr.role]="isHandset ? 'dialog' : 'navigation'"
    [mode]="(isHandset | async)!.matches ? 'over' : 'side'"
    [opened]="!(isHandset | async)!.matches">
    <mat-toolbar color="primary">Menu</mat-toolbar>
    <mat-nav-list>
      <a mat-list-item href="">First Page</a>
      <a mat-list-item href="">Second Page</a>
      <a mat-list-item href="">Third Page</a>
    </mat-nav-list>
  </mat-sidenav>
  <mat-sidenav-content>
    <mat-toolbar color="primary">
      <button
        type="button"
        aria-label="Toggle sidenav"
        mat-icon-button
        (click)="drawer.toggle()"
        *ngIf="(isHandset | async)!.matches">
        <mat-icon aria-label="Side nav toggle icon">menu</mat-icon>
      </button>
      <span>Application Title</span>
    </mat-toolbar>
    <router-outlet></router-outlet>
  </mat-sidenav-content>
</mat-sidenav-container>

**Comment by amcdnl** [comment]
Addressed the height issue in #11191 ... otherwise looks good on 6.2.0

**Comment by Splaktar** [comment]
Fixed in 6.2.0 via https://github.com/angular/material2/commit/b9041e34cceafe2a7011ea7bd42d4fa167819266.
Overlay issue fixed in 6.3.0 via https://github.com/angular/material2/commit/6bd2775e25c2fc8b8c4376edeb7f7dc7783e7e4a.
